prow/plugins/verify-owners/verify-owners.go

```diff
@@ -126,27 +155,107 @@ func handle(ghc githubClient, gc *git.Client, log *logrus.Entry, pre *github.Pul
 		}
 	}
 
-	// Check each OWNERS file.
+	// If OWNERS_ALIASES file exists, get all aliases.
+	path := filepath.Join(r.Dir, ownersAliasesFileName)
+	if _, err := os.Stat(path); err == nil {
+		b, err := ioutil.ReadFile(path)
+		if err != nil {
+			log.WithError(err).Errorf("Failed to read %s.", path)
+			return nil
+		}
+		repoAliases, err = repoowners.ParseAliasesConfig(b)
+		if err != nil {
+			return fmt.Errorf("error parsing aliases config for %s file: %v", ownersAliasesFileName, err)
+		}
+
+		// If OWNERS_ALIASES file was modified, check if newly added owners are collaborators.
+		if ownerAliasesModified {
+			allOwners := repoAliases.ExpandAllAliases().List()
+			for _, owner := range allOwners {
+				// cap the number of checks to avoid exhausting tokens in case of large OWNERS refactors.
+				if len(nonCollaborators) > 20 {
+					break
+				}
+
+				// look for newly added owners by checking if they are in the patch.
+				if strings.Contains(modifiedOwnerAliasesFile.Patch, owner) {
+					isCollaborator, err := ghc.IsCollaborator(org, repo, owner)
+					if err != nil {
+						log.WithError(err).Errorf("Failed IsCollaborator(%s, %s, %s)", org, repo, owner)
+					}
+
+					if !isCollaborator {
+						if ownersFiles, ok := nonCollaborators[owner]; ok {
+							nonCollaborators[owner] = append(ownersFiles, ownersAliasesFileName)
+						} else {
+							nonCollaborators[owner] = []string{ownersAliasesFileName}
+						}
+					}
+				}
+			}
+		}
+	}
+
+	// Check if OWNERS files have the correct config and if they do,
+	// check if all newly added owners are collaborators.
 	for _, c := range modifiedOwnersFiles {
-		// Try to load OWNERS file.
 		path := filepath.Join(r.Dir, c.Filename)
 		b, err := ioutil.ReadFile(path)
 		if err != nil {
 			log.WithError(err).Warningf("Failed to read %s.", path)
 			return nil
 		}
-		if msg := parseOwnersFile(b, c, log, labelsBlackList); msg != nil {
+		msg, owners := parseOwnersFile(b, c, log, labelsBlackList)
+		if msg != nil {
 			wrongOwnersFiles[c.Filename] = *msg
+		} else {
+			// the OWNERS file has the correct config, check for non-collaborators now.
+			for _, owner := range owners {
+				// look for newly added owners by checking if they are in the patch.
+				if strings.Contains(c.Patch, owner) {
+					// cap the number of checks to avoid exhausting tokens in case of large OWNERS refactors.
+					if len(nonCollaborators) > 20 {
+						break
+					}
+
+					// check if owner is actually an alias.
+					if _, ok := repoAliases[owner]; ok {
```

Comment:
IMO we should check that everyone in the alias is an owner

Reply:
> IMO we should check that everyone in the alias is an owner

This is done while checking owners in OWNERS_ALIASES (`nonTrustedUsersInOwnersAliases`)